conda_env/cli/main.py

```diff
@@ -11,34 +11,33 @@
 # when importing pip (and pip_util)
 import conda.exports  # noqa
 from conda.base.context import context
+from conda.cli import (
+    main_env_config,
+    main_env_create,
+    main_env_export,
+    main_env_list,
+    main_env_remove,
+    main_env_update,
+)
 from conda.cli.conda_argparse import ArgumentParser
+from conda.cli.helpers import show_help_on_empty_command
 from conda.cli.main import init_loggers
 from conda.exceptions import conda_exception_handler
 from conda.gateways.logging import initialize_logging
 
-from . import main_config, main_create, main_export, main_list, main_remove, main_update
-
-
-# TODO: This belongs in a helper library somewhere
-# Note: This only works with `conda-env` as a sub-command.  If this gets
-# merged into conda-env, this needs to be adjusted.
-def show_help_on_empty_command():
-    if len(sys.argv) == 1:  # sys.argv == ['/path/to/bin/conda-env']
-        sys.argv.append("--help")
-
 
 def create_parser():
     p = ArgumentParser()
     sub_parsers = p.add_subparsers(
         metavar="command",
         dest="cmd",
     )
-    main_create.configure_parser(sub_parsers)
-    main_export.configure_parser(sub_parsers)
-    main_list.configure_parser(sub_parsers)
-    main_remove.configure_parser(sub_parsers)
-    main_update.configure_parser(sub_parsers)
-    main_config.configure_parser(sub_parsers)
+    main_env_create.configure_parser(sub_parsers)
+    main_env_export.configure_parser(sub_parsers)
+    main_env_list.configure_parser(sub_parsers)
+    main_env_remove.configure_parser(sub_parsers)
+    main_env_update.configure_parser(sub_parsers)
+    main_env_config.configure_parser(sub_parsers)
 
     show_help_on_empty_command()
     return p
```

Comment:
Currently when we call `conda env` the following code path is triggered:

1. conda entrypoint (`$CONDA_PREFIX/bin/conda`)
2. `conda.cli.main.main()`
3. `conda.cli.main.main_subshell()`
4. `conda.cli.conda_argparse.do_call()`, second if-clause: https://github.com/conda/conda/blob/def9835a389e43dfad0afbabab6adeefd49b8c12/conda/cli/conda_argparse.py#L172-L185


We actually want this implemented and called as a first class plugin so this `create_parser` needs to be reimplemented as a `configure_parser` function in a new submodule `conda.cli.main_env`.